pkg/query/api/v1.go

```diff
@@ -163,7 +164,7 @@ func NewAPI(
 }
 
 // Register the API's endpoints in the given router.
-func (api *API) Register(r *route.Router, tracer opentracing.Tracer, logger log.Logger, ins extpromhttp.InstrumentationMiddleware) {
+func (api *API) Register(r *route.Router, tracer opentracing.Tracer, logger log.Logger, ins extpromhttp.InstrumentationMiddleware, logMiddleware *logging.LogServerMiddleware) {
```

Comment:
Since we are touching this method, let's create a generic middleware interface and convert middleware args to varargs.

Reply:
More info - https://github.com/thanos-io/thanos/issues/2914